Assets/MRTK/Core/Services/MixedRealityToolkit.cs

```diff
@@ -1463,7 +1463,8 @@ static EditorEventListener()
         /// Used to register newly created instances in edit mode.
         /// Initially handled by using ExecuteAlways, but this attribute causes the instance to be destroyed as we enter play mode, which is disruptive to services.
         /// </summary>
-        private void OnValidate()
+        [UnityEditor.Callbacks.DidReloadScripts] // This is workaround for a known unity issue when calling refresh assetdatabase from inside a on validate scope.
```

Comment:
While testing this out, it seems like this gets called even when no MixedRealityToolkit script exists in the scene. This leads to:

```
Exception thrown while invoking [DidReloadScripts] method 'Microsoft.MixedReality.Toolkit.MixedRealityToolkit:OnDidReloadScripts ()' : NullReferenceException: Object reference not set to an instance of an object
Microsoft.MixedReality.Toolkit.MixedRealityToolkit.RegisterInstance (Microsoft.MixedReality.Toolkit.MixedRealityToolkit toolkitInstance, System.Boolean setAsActiveInstance) (at Assets/MRTK/Core/Services/MixedRealityToolkit.cs:720)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit.OnDidReloadScripts () (at Assets/MRTK/Core/Services/MixedRealityToolkit.cs:1478)
```

`OnValidate` (I believe) only gets called on things actually present in the inspector, so it sorta guarantees that something will exist.

Reply:
Nice catch Kurtis.